src/lib/ares_library_init.c

```diff
@@ -45,6 +45,14 @@ static int          ares_init_flags;
 static void *default_malloc(size_t size) { return malloc(size); }
 static void *default_realloc(void *p, size_t size) { return realloc(p, size); }
 static void default_free(void *p) { free(p); }
+#elif defined(__MVS__)
```

Comment:
This block should be removed.  If there are really any malloc(0) issues in c-ares, those need to be fixed instead.